monitoring/uss_qualifier/scd/simulator/operational_intents_flight_authorisation_data_generator.py

```diff
@@ -0,0 +1,371 @@
+        reference_time = my_operational_intent_generator.now.isoformat()
+    
+        flight_volume = my_operational_intent_generator.generate_astm_4d_volumes(raw_geometry = flight_geometry, altitude_of_ground_level_wgs_84 = altitude_of_ground_level_wgs_84)
+        
+        operational_intent_test_injection = generate_injection_operational_intent(astm_4d_volume = flight_volume)
+    
+        inject_flight_request = InjectFlightRequest(operational_intent= operational_intent_test_injection, flight_authorisation= flight_authorisation_data)
+        
+        expected_flight_authorisation_processing_result = 'Rejected' if generate_incorrect_data else 'Planned'
+        expected_operational_intent_processing_result= 'ConflictWithFlight' if should_intersect else 'Planned'
+
+        my_known_issues_acceptable_result_generator = KnownIssuesAcceptableResultFieldGenerator(expected_flight_authorisation_processing_result = expected_flight_authorisation_processing_result,expected_operational_intent_processing_result = expected_operational_intent_processing_result)
+        all_incorrect_result_details = my_known_issues_acceptable_result_generator.generate_known_issue_fields(incorrect_field= incorrect_field)
+
+        acceptable_results = my_known_issues_acceptable_result_generator.acceptable_results
+        
+        injection_target = InjectionTarget(uss_role = "Submitting USS")
+        
+        known_responses = KnownResponses(acceptable_results=acceptable_results, incorrect_result_details= all_incorrect_result_details)
+        
+        flight_injection_attempt = FlightInjectionAttempt(reference_time = reference_time, test_injection = inject_flight_request, known_responses = [known_responses],injection_target = injection_target)
+        
+        flight_injection_attempts.append(flight_injection_attempt)
+    
+    return flight_injection_attempts
+    
+
+def write_automated_test_to_disk(output_path:os.path, flight_injection_attempts: List[FlightInjectionAttempt], country_code="che") -> None:
+    """A function to write Flight injection attempts to disk so that they can be examined / used by other software like the test executor """
+        
+    output_directory = Path(output_path, country_code) 
+    # Create test_definition directory if it does not exist        
+    output_directory.mkdir(parents=True, exist_ok=True)        
+    output_subdirectories = Path(output_directory, "automated_test")        
+    output_subdirectories.mkdir(parents=True, exist_ok=True)       
+
+    automated_test_data = AutomatedTest(name="Nominal Strategic co-ordination", injection_attempts = flight_injection_attempts)
+        
+    automated_test_file_name = "automated_test_%s.json" % str(1)  # Avoid Zero based numbering           
+    automated_test_file = Path(output_subdirectories, automated_test_file_name)
+    with open(automated_test_file, "w") as f:
+        f.write(json.dumps(automated_test_data))
```

Comment:
Following this line, let's check accuracy by adding:

```python
with open(automated_test_file, 'r') as f:
    ImplicitDict.parse(json.load(f), AutomatedTest)
```

Reply:
Thanks! I have now added the check

Reply:
#700 should fix those issues